apps/dashboard/src/app/nebula-app/(app)/api/fetchWithAuthToken.ts

```diff
@@ -0,0 +1,50 @@
+type FetchWithKeyOptions = {
+  endpoint: string;
+  authToken: string;
+  timeout?: number;
+} & (
+  | {
+      method: "POST" | "PUT";
+      body: Record<string, unknown>;
+    }
+  | {
+      method: "GET" | "DELETE";
+    }
+);
+
+export async function fetchWithAuthToken(options: FetchWithKeyOptions) {
```

Comment:
this seems useful generally also for regular dashboard, maybe move it u[?

Reply:
yeah, sure

Reply:
done